src/database/db.js

```diff
@@ -34,56 +36,120 @@ const Clip = mongoose.model('Clip', ClipSchema);
 // Database operations
 const db = {
   // Store a new clip
-  storeClip: async (url, description, submittedBy) => {
-    const newClip = new Clip({ url, description, submittedBy });
-    return await newClip.save();
+  storeClip: async (url, description, submittedBy, interactionId, messageId) => {
+    try {
+      const newClip = new Clip({ 
+        url, 
+        description, 
+        submittedBy, 
+        interactionId, 
+        messageId 
+      });
+      return await newClip.save();
+    } catch (error) {
+      console.error('Error storing clip:', error);
+      throw error;
+    }
   },
 
   // Add a rating to a clip
   addRating: async (clipId, rating, ratedBy) => {
-    return await Clip.findByIdAndUpdate(
-      clipId,
-      { $push: { ratings: { rating, ratedBy } } },
-      { new: true }
-    );
+    try {
+      return await Clip.findByIdAndUpdate(
+        clipId,
+        { $push: { ratings: { rating, ratedBy } } },
+        { new: true }
+      );
```

Comment:
Does this allow users to update ratings, or only assign new ones?

Reply:
This only allows users to add new ratings:

**Key points:**
1. It always adds a new rating object to the array.
2. It doesn't check if the user (ratedBy) has already rated the clip.
3. There's no mechanism to update an existing rating.

Should we update this to allow updates?

Reply:
Yes, either that or add a new method to update ratings as we should expect users to be able to change their ratings